### PR TITLE
Add dependency on leak_tracker to flutter_test.

### PR DESCRIPTION
**Pull request by polina-c**
Contributes to: https://github.com/flutter/flutter/issues/135856

This PR also adds transitive dependency on  web_socket_channel, crypto, typed_data. `web_socket_channel` is needed to request retaining path for not GCed objects from VM Service. Two others are needed to serve web_socket_channel.

The dependency on leak_tracker is pinned:
https://github.com/flutter/flutter/blob/aea562114c4d8c220c9079e82d89a7da4164409f/packages/flutter_tools/lib/src/commands/update_packages.dart#L40

### CODE REVIEW

File: packages/flutter/pubspec.yaml [hunk 1]
```diff
@@ -22,17 +22,16 @@ dev_dependencies:
   flutter_goldens:
     sdk: flutter
   fake_async: 1.3.1
-  leak_tracker: 9.0.8
-  leak_tracker_flutter_testing: 1.0.6
+  leak_tracker: 9.0.11
+  leak_tracker_testing: 1.0.5
+  leak_tracker_flutter_testing: 1.0.9
```
Comment (dnfield) [inline packages/flutter/pubspec.yaml:27]: Do we really want to have runtime dependencies on these packages in package/flutter?

Comment (polina-c) [reply packages/flutter/pubspec.yaml:27]: First, they are just dev_dependencies.
Second, they will go away after we clean up `testWidgetsWithLeakTracking` from package/flutter/test.

**Review by jonahwilliams — APPROVED** [review]
LGTM

File: packages/flutter/pubspec.yaml (hunk 1 above)
Comment (dnfield) [reply packages/flutter/pubspec.yaml:27]: Ahh I see now, missed this the way the GH diff was collapsed.

**Review by dnfield — APPROVED** [review]
LGTM

**Pull request merged**